src/app/betting/feature/place-bet-page.component.ts

```diff
@@ -219,8 +477,74 @@ export default class PlaceBetPageComponent {
     });
   }
 
-  // When player list changes (e.g. someone's score changes externally)
-  // clear selected opponent so it doesn't get orphaned
+  generateBetTypeObject(type: BetType) {
+    switch (type) {
+      case BetType.DuelWinner:
+        return { betType: type, betTypeString: 'Duel Winner' };
+      case BetType.SpecialSpaceEvent:
+        return {
+          betType: type,
+          betTypeString: 'Special Space Event Over/Under',
+        };
+      default:
+        return { betType: BetType.Manual, betTypeString: 'Manual' };
+    }
+  }
+
+  private generateDetails(): Json {
```

Comment:
This is getting nitpicky, so feel free to ignore this comment if you want.

I would rather see this function (and similar functions) be "pure functions," meaning they take in all dependencies as arguments rather than relying on `this.whatever` inside the function body. And with functions like this, I think it makes sense for them to be standalone functions that exist outside the service (either at the bottom of this file or in another file).

Here's how this function could be updated:

```typescript
function generateBetDetails<T extends BetType>(betType: T, duel: DuelModel | undefined, /* etc... */): BetDetails<T> {
  switch (betType) {
    case BetType.DuelWinner:
      return {
        duelId: selectedDuel?.id,
        challengerWins:
          selectedWinner?.player_id ===
          selectedDuel?.challenger?.player_id,
      };
    case BetType.SpecialSpaceEvent:
      return {
        ssEventId: selectedSsEvent?.ssEvent?.id,
        directionIsOver: selectedOuOption === 'Over',
        ouValue,
      };
    default:
      return {};
  }
}
```

Here's what we could do for strongly-typed bet details:

```typescript
interface DuelWinnerBetDetails {
  duelId: DuelModel['id'];
  challengerWins: boolean;
}

interface SpecialSpaceEventBetDetails {
  ssEventId: SpecialSpaceEventModel['id'];
  directionIsOver: boolean;
  ouValue: number;
}

type BetDetails<T extends BetType = BetType> =
  T extends BetType.DuelWinner ?
    DuelWinnerBetDetails :
      T extends BetType.SpecialSpaceEvent ?
        SpecialSpaceEventBetDetails :
        {};
```